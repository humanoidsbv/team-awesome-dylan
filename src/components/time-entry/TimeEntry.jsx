import React from 'react';

import styles from './TimeEntry.module.css';

const TimeEntry = () => (
  <div className={styles.timeEntry}>
    <span className={styles.client}>Port of Rotterdam</span>
    <div className={styles.timeStamp}>
      <span>09:00 - 17:00</span>
      <span className={styles.duration}>
        <br />
        08:00:00
      </span>
    </div>
  </div>
);

export default TimeEntry;
