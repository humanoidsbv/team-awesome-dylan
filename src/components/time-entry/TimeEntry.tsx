import React from 'react';

import IconDelete from '../../assets/icons/icon-delete.svg';
import styles from './TimeEntry.module.css';
import { TimeEntryInterface } from '../../redux/time-entries/types';

interface TimeEntryProps extends TimeEntryInterface {
  client: string;
  deleteTimeEntry: (number) => {};
}

const TimeEntry = ({
  client, deleteTimeEntry, id, startTimestamp, stopTimestamp
}: TimeEntryProps): React.ReactElement => {
  const clientDurationParse = (
    Date.parse(stopTimestamp) - Date.parse(startTimestamp)) / 60 / 60 / 1000;

  const durationHours = Math.trunc(clientDurationParse);
  const durationMinutes = Math.round((clientDurationParse % 1) * 60);
  const totalClientDuration = `${durationHours}:${durationMinutes
    .toString()
    .padStart(2, '0')}`;

  const timeNotation = { hour: '2-digit', minute: '2-digit' };

  const startTimeString = new Date(startTimestamp).toLocaleTimeString(
    'nl-NL',
    timeNotation
  );

  const stopTimeString = new Date(stopTimestamp).toLocaleTimeString(
    'nl-NL',
    timeNotation
  );

  return (
    <div className={styles.timeEntry}>
      <span className={styles.client}>{client}</span>
      <button
        className={styles.deleteButton}
        onClick={(): {} => deleteTimeEntry(id)}
        type="button"
      >
        <IconDelete className={styles.iconClose} />
        Delete
      </button>
      <div className={styles.timeStamp}>
        <span>{`${startTimeString} - ${stopTimeString}`}</span>
        <span className={styles.duration}>
          <br />
          {totalClientDuration}
        </span>
      </div>
    </div>
  );
};

// TimeEntry.propTypes = {
//   id: PropTypes.number.isRequired,
//   client: PropTypes.string.isRequired,
//   deleteTimeEntry: PropTypes.func.isRequired,
//   startTime: PropTypes.string.isRequired,
//   stopTime: PropTypes.string.isRequired
// };

export default TimeEntry;
