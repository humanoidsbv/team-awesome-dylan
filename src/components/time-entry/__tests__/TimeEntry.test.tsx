import * as React from 'react';
import { shallow } from 'enzyme';

import TimeEntry from '../TimeEntry';

it('converts basic button using shallow', () => {
  const timeEntry = shallow(
    <TimeEntry
      id={123}
      client={{
        branch: '123',
        clientName: '123',
        id: 123,
        locality: '123',
        postalCode: '123',
        contactPerson: 'test',
        contactPhoneNumber: 'test'
      }}
      deleteTimeEntry={(id: number) => null}
      startTimestamp="2019-10-16T10:00:00.000Z"
      stopTimestamp="2019-10-16T17:30:00.000Z"
    />
  );
  expect(timeEntry).toMatchSnapshot();
});
