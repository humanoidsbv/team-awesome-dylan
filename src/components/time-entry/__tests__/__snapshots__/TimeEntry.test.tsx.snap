// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="timeEntry"
>
  <span
    className="client"
  >
    123
  </span>
  <button
    className="deleteButton"
    onClick={[Function]}
    type="button"
  >
    <test-file-stub
      className="iconClose"
    />
    Delete
  </button>
  <div
    className="timeStamp"
  >
    <span>
      12:00 - 19:30
    </span>
    <span
      className="duration"
    >
      <br />
      7:30
    </span>
  </div>
</div>
`;
